### PR TITLE
feat: Add GraphLegend

### LINKED ISSUE #3515: Graph2 Legend
With the new Graph2 "categories" and "annotations", we should be able to auto generate a sensible legend.

I would suggest making it a different widget but with the same "categories" and "annotations" methods.

The inputs will probably need extending to include a description (which is also useful for the core Graph control.

The one we generate in ECL Watch looks like this:
image

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #3515

Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [ ] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by jbrundage** [comment]
Instantiation is simple: https://github.com/hpcc-systems/Visualization/pull/3519/files#diff-ecef7030f291bcc487d44af1523440dcR194-R198

**Comment by jbrundage** [comment]
image

File: packages/graph/src/graph2/legend.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            .hierarchyRankDirection("LR")
+            .hierarchyNodeSeparation(30)
+            .allowDragging(false)
+            .highlightOnMouseOverVertex(false)
+            ;
+    }
+
+    update(domNode, element) {
+
+        const dataHasChanged = true;
```
Comment (jbrundage) [inline packages/graph/src/graph2/legend.ts]: @GordonSmith Is this check worth implementing? It seems like the only time we'd call render on GraphLegend is when the data changes anyway.

File: packages/graph/src/graph2/legend.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            this.data({
+                vertices,
+                edges: []
+            });
+        }
+
+        super.update(domNode, element);
+    }
+}
+GraphLegend.prototype._class += " graph_GraphLegend";
```
Comment (GordonSmith) [inline packages/graph/src/graph2/legend.ts]: * Shouldn't derive from Graph2.
* Should have `categories` and `annotations` methods.
* Would like to see a demo of it working in a Border2 with a Graph in the center.

File: packages/graph/src/graph2/legend.ts (hunk 1 above)
Comment (GordonSmith) [reply packages/graph/src/graph2/legend.ts]: Agree - and the data is "small"

**Comment by GordonSmith** [comment]
@jbrundage bump

**Comment by GordonSmith** [comment]
@jbrundage bump (and it needs rebasing)

**Comment by GordonSmith** [comment]
@jbrundage bump 3 (and it needs rebasing)

File: packages/graph/src/graph2/legend.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/graph/src/graph2/legend.ts:76]: @jbrundage bump

Comment (jbrundage) [reply packages/graph/src/graph2/legend.ts:76]: Yes, I have a nearly finished version of these requested changes that I need to go back to

**Pull request closed without merging**